elevenlabs/api/voice.py

```diff
@@ -137,6 +137,32 @@ def computed_settings(cls, v: VoiceSettings, values) -> VoiceSettings:
     def delete(self):
         API.delete(f"{api_base_url_v1}/voices/{self.voice_id}")
 
+    def edit_settings(self, stability: float, similarity_boost: float):
+        url = f"{api_base_url_v1}/voices/{self.voice_id}/settings/edit"
+        json = {
+            "stability": stability,
+            "similarity_boost": similarity_boost,
+        }
+        API.post(url, json=json)
+
+    def get_settings(self):
+        url = f"{api_base_url_v1}/voices/{self.voice_id}/settings"
+        return VoiceSettings(**API.get(url).json())
+    
+    def get_info(self):
+        url = f"{api_base_url_v1}/voices/{self.voice_id}"
+        return Voice(**API.get(url).json())
+    
+    def edit(self, name: str, labels: Optional[str] = None, description: Optional[str] = None):
```

Comment:
This can get out of sync with the `self` object, e.g. name can be different from the one in the API. I'd do something like 
```py 
self.name = name or self.name,
self.labels = labels or self.labels,
self.description = description or self.description

data = dict(name=self.name, labels=self.labels, description=self.description) 
```

where also the `name` input is optional. This also allows the user to manually change the voice object, e.g. `voice.name='new name',` and then call `voice.edit()`

Reply:
Thanks for bringing that to my attention, I agree with you and I'll implement these changes. However, pardon my question, but did you mean that you want me to use `cls` instead of `self`? Also, I was wondering, could you tell me what the purpose of `@validator("settings")` is? I never saw such a thing  before.

Reply:
Depends on whether you want to modify the current object or any voice object using it's id, if you want you can do both `edit(self, name, labels, ...)` to edit the instance object and `edit(cls, voice_id, name,...)` which can be called as `Voice.edit(voice_id, name, ...)`. 

The `@validator("settings")` basically calls the `computed_settings` function when the voice object is created, this is to do some custom initialization of the `settings` field. The function provides as input the current value of the `settings: VoiceSettings` when the class is initiazlied, then you can do whatever you want to it and return the updated value. In this case, since the API doesn't return the settings of the voice when doing `Voice.from_id(..)`, we use this to automatically fetch the /settings endpoint, and populate the `settings` field. If instead you manually instantiate a voice object with `Voice(settings=VoiceSettings(...))` the input to that function will not be `None` and hence nothing will happen: `v if v else VoiceSettings(**API.get(url).json())` (see the `if` check)